Author: Andreas Beckmann <anbe@debian.org>
Description: select the correct nv-modeset-kernel.o blob for the target architecture

--- a/nvidia-modeset/nvidia-modeset.Kbuild
+++ b/nvidia-modeset/nvidia-modeset.Kbuild
@@ -35,7 +35,9 @@ NV_KERNEL_MODULE_TARGETS += $(NVIDIA_MOD
 # But, the target for the symlink rule should be prepended with $(obj).
 #
 
-NVIDIA_MODESET_BINARY_OBJECT := $(src)/nvidia-modeset/nv-modeset-kernel.o_binary
+NVIDIA_MODESET_BINARY_OBJECT-$(CONFIG_X86_64)	+= nv-modeset-kernel-amd64.o_binary
+NVIDIA_MODESET_BINARY_OBJECT-$(CONFIG_PPC64)	+= nv-modeset-kernel-ppc64el.o_binary
+NVIDIA_MODESET_BINARY_OBJECT := $(src)/nvidia-modeset/$(NVIDIA_MODESET_BINARY_OBJECT-y)
 NVIDIA_MODESET_BINARY_OBJECT_O := nvidia-modeset/nv-modeset-kernel.o
 
 quiet_cmd_symlink = SYMLINK $@
